katran/lib/bpf/balancer_kern.c

```diff
@@ -346,27 +314,18 @@ static inline int process_packet(void *data, __u64 off, void *data_end,
     }
     __u32 *decap_dst_flags = bpf_map_lookup_elem(&decap_dst, &dst_addr);
 
-    action = process_encaped_pckt(&data, &data_end, xdp, &is_ipv6, &pckt,
-                                  &protocol, off, &pkt_bytes, decap_dst_flags);
-    if (action >= 0) {
-      return action;
-    }
-
     if (decap_dst_flags) {
       __u32 stats_key = MAX_VIPS + REMOTE_ENCAP_CNTRS;
       data_stats = bpf_map_lookup_elem(&stats, &stats_key);
       if (!data_stats) {
         return XDP_DROP;
       }
       data_stats->v1 += 1;
-      pckt.flags |= F_INLINE_DECAP;
-    } else {
-      // it's ipip encapsulated packet but not to decap dst. so just pass
-      // decapsulated packet to the kernel
-      return XDP_PASS;
     }
+
+    return process_encaped_pckt(&data, &data_end, xdp, &is_ipv6, &protocol);
```

Comment:
With INLINE_DECAP + Gue - we'll keep on looping till ttl gets to 0?

Reply:
line 308. for GUE this code path wont be triggered. as it only works for ipip/ip6ip6.